src/Control/Monad/Aff.purs

```diff
@@ -65,9 +65,12 @@ module Control.Monad.Aff
   cancelWith aff c = runFn3 _cancelWith nonCanceler aff c
 
   -- | Converts the asynchronous computation into a synchronous one. All values
-  -- | and errors are ignored.
-  launchAff :: forall e a. Aff e a -> Eff e Unit
-  launchAff = runAff (const (pure unit)) (const (pure unit))
+  -- | are ignored, and if the computation produces an error, it is thrown.
+  launchAff :: forall e a. Aff e a -> Eff (err :: EXCEPTION | e) Unit
```

Comment:
After mulling this over the past couple days, I think the best option right now is to leave it as is, but add the comment. This way you can still 'catch' a synchronous exception (which may in fact happen, since `Aff` is not *necessarily* asynchronous), while the comment will warn you away from assuming that future exceptions can be caught synchronously.

Now what we'd really like is for all `main` functions to expect `Aff` and not `Eff`. :wink: